src/spyglass/decoding/v0/dj_decoder_conversion.py

```diff
@@ -0,0 +1,201 @@
+"""Converts decoder classes into dictionaries and dictionaries into classes
+so that datajoint can store them in tables."""
+from spyglass.utils import logger
+
+try:
+    from replay_trajectory_classification.continuous_state_transitions import (
+        Identity,
+        RandomWalk,
+        RandomWalkDirection1,
+        RandomWalkDirection2,
+        Uniform,
+    )
+    from replay_trajectory_classification.discrete_state_transitions import (
+        DiagonalDiscrete,
+        RandomDiscrete,
+        UniformDiscrete,
+        UserDefinedDiscrete,
+    )
+    from replay_trajectory_classification.environments import Environment
+    from replay_trajectory_classification.initial_conditions import (
+        UniformInitialConditions,
+        UniformOneEnvironmentInitialConditions,
+    )
+    from replay_trajectory_classification.observation_model import (
+        ObservationModel,
+    )
+    from track_linearization import make_track_graph
+
+    def _convert_dict_to_class(d: dict, class_conversion: dict) -> object:
+        """Converts a dictionary into a class object
+
+        Parameters
+        ----------
+        d : dict
+        class_conversion : dict
+
+        Returns
+        -------
+        class_based_on_dict : object
+
+        """
+        class_name = d.pop("class_name")
+        return class_conversion[class_name](**d)
+
+    def _convert_env_dict(env_params: dict) -> Environment:
+        """If the track graph is in the environment parameters, convert it to a networkx graph
+
+        Parameters
+        ----------
+        env_params : dict
+
+        Returns
+        -------
+        environment : Environment
+        """
+        if env_params["track_graph"] is not None:
+            env_params["track_graph"] = make_track_graph(
+                **env_params["track_graph"]
+            )
+
+        return Environment(**env_params)
+
+    def _to_dict(transition: object) -> dict:
+        """Helper function to convert a transition class into a dictionary"""
+        parameters = vars(transition)
+        parameters["class_name"] = type(transition).__name__
+
+        return parameters
+
+    def _convert_transitions_to_dict(
+        transitions: list[list[object]],
+    ) -> list[list[dict]]:
+        """Converts a list of lists of transition classes into a list of lists of dictionaries"""
+        return [
+            [_to_dict(transition) for transition in transition_rows]
+            for transition_rows in transitions
+        ]
+
+    def restore_classes(params: dict) -> dict:
+        """Converts a dictionary of parameters into a dictionary of classes since datajoint cannot handle classes"""
+        continuous_state_transition_types = {
```

Comment:
Should these class dicts be script constants? Do you anticipate that more classes may be added in the future? Whether or not you want to be explicit about the class list, you can use a dict comprehension:
```
from replay_trajectory_classification import continuous_state_transitions as cst
module_cst_attributes = dir(cst)
continuous_state_transition_types = {
    attr_name: attr
    for attr_name, attr in [(name, getattr(cst, name)) for name in module_cst_attributes]
    if hasattr(attr, "__class__") and attr.__class__.__name__ == "type"
}
```